Sources/TSPL/TSPL.docc/ReferenceManual/Attributes.md

```diff
@@ -1362,11 +1364,38 @@ If your result-building methods
 don't specify a type for `Expression` or `FinalResult`,
 they default to being the same as `Component`.
 
-The result-building methods are as follows:
+The block-building methods are as follows:
 
 - term `static func buildBlock(_ components: Component...) -> Component`:
   Combines an array of partial results into a single partial result.
-  A result builder must implement this method.
+
+- term `static func buildPartialBlock(first: Component) -> Component`:
+  Builds a partial result component from the first component.
+  Implement both this method and `buildPartialBlock(accumulated:next:)`
+  to support building blocks one component at a time.
+  Compared to `buildBlock(_:)`,
+  this approach reduces the need for generic overloads
+  that handle different numbers of arguments.
+
+- term `static func buildPartialBlock(accumulated: Component, next: Component) -> Component`:
+  Builds a partial result component
+  by combining an accumulated component with a new component.
+  Implement both this method and `buildPartialBlock(first:)`
+  to support building blocks one component at a time.
+  Compared to `buildBlock(_:)`,
+  this approach reduces the need for generic overloads
+  that handle different numbers of arguments.
+
+A result builder can implement
+all three of the block-building methods listed above ---
+in that case,
+availability determines which method is used.
+If the enclosing declaration is available
+before the build-partial-block methods,
```

Comment:
@tshortli is there a better unavailability phrasing than this i.e. "more broadly available" or something?

Reply:
I often say "more available" to describe this relationship; a type that is available on macOS 11 is more available than a type available on macOS 12

Reply:
After some discussion with DevPubs Editorial, I'm using the following phrasing:

A result builder can implement all three of the block-building methods listed above; in that case, availability determines which method is called.  By default, Swift calls the `buildPartialBlock(first:)` and `buildPartialBlock(second:)` methods.  To make Swift call `buildBlock(_:)` instead, mark the enclosing declaration as being available before the availability you write on `buildPartialBlock(first:)` and `buildPartialBlock(second:)`.

Reply:
Sound good to me!